### PR TITLE
README rewrite for DU functions

### PR DESCRIPTION
**Pull request by jtakalai**
Updated the DU function docs to reflect the fact that we now have the DataUnion object (examples rewritten)

Also general cleanup. Please add freely to it where you see shortcomings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -339,19 +341,27 @@ All the below functions return a Promise which gets resolved with the result.
 
 ## Data Unions
 
-This library provides functions for working with Data Unions. To get a DataUnion instance, call `client.getDataUnion(address)`. To deploy a new DataUnion, call `deployDataUnion(options)`
+This library provides functions for working with Data Unions.
 
-See `/docs` folder in this repo for Data Union `options` parameters. These docs can be rebuilt locally via:
+To deploy a new DataUnion:
```
Comment (fonty1) [inline README.md:346]: Is this mixed up with 
**_To get an existing (previously deployed) `DataUnion` instance:_**

File: README.md [hunk 2]
```diff
@@ -367,12 +377,14 @@ These DataUnion-specific options are used from `StreamrClient` options:
 Here's an example how to deploy a data union contract and set the admin fee:
```
Comment (fonty1) [inline README.md:377]:
```suggestion
Here's how to deploy a Data Union contract and set the admin fee to 30%:
```

File: README.md [hunk 3]
```diff
@@ -339,19 +341,27 @@ All the below functions return a Promise which gets resolved with the result.
 
 ## Data Unions
 
-This library provides functions for working with Data Unions. To get a DataUnion instance, call `client.getDataUnion(address)`. To deploy a new DataUnion, call `deployDataUnion(options)`
+This library provides functions for working with Data Unions.
 
-See `/docs` folder in this repo for Data Union `options` parameters. These docs can be rebuilt locally via:
+To deploy a new DataUnion:
+```js
+const dataUnion = client.getDataUnion(address)
```
Comment (teogeb) [inline README.md:348]: This example codes and their descriptions don't match

File: README.md [hunk 4]
```diff
@@ -367,12 +377,14 @@ These DataUnion-specific options are used from `StreamrClient` options:
...
 ```js
+import { StreamrClient } from 'streamr-client'
+
 const client = new StreamrClient({
     auth: { privateKey },
 })
 
 const dataUnion = await client.deployDataUnion()
-await client.setAdminFee(0.3, { dataUnion })
+const receipt = await dataUnion.setAdminFee(0.3, { dataUnion })
```
Comment (teogeb) [inline README.md:387]: The second parameter `(({ dataUnion })` has been removed

**Review by teogeb — COMMENTED** [review]
Good improvements! Could also say that DU methods returns their values in Promises if that is not 100% clear.

File: README.md (hunk 1 above)
Comment (jtakalai) [reply README.md:346]: good catch

File: README.md (hunk 3 above)
Comment (jtakalai) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (jtakalai) [reply README.md:387]: it has in fact been removed  ;)  https://github.com/streamr-dev/streamr-client-javascript/pull/224

Comment (jtakalai) [reply README.md:387]: or about to be

Comment (teogeb) [reply README.md:387]: In #224 you remove option for another method (`addMembers`, `removeMembers`), not `setAdminFee`.

File: README.md [hunk 5]
```diff
@@ -339,62 +341,87 @@ All the below functions return a Promise which gets resolved with the result.
...
+Later, anyone (e.g. Data Union admin) can send that withdraw transaction to the blockchain (and pay for the gas)
+
+```js
+import { StreamrClient } from 'streamr-client'
+
+const client = new StreamrClient({
+    auth: { privateKey },
+})
+
+const dataUnion = client.getDataUnion(dataUnionAddress)
```
Comment (fonty1) [inline README.md:423]:
```suggestion
const dataUnion = client.getDataUnion(address)
```
for consistency,  dataUnionAddress -> address

File: README.md [hunk 6]
```diff
@@ -413,19 +440,33 @@ These are available for everyone and anyone, to query publicly available info fr
 Here's an example how to get a member's withdrawable token balance (in "wei", where 1 DATA = 10^18 wei)
 
 ```js
-const client = new StreamrClient({
-    dataUnion,
-})
+import { StreamrClient } from 'streamr-client'
 
-const withdrawableWei = await client.getMemberBalance(memberAddress)
+const dataUnion = new StreamrClient().getDataUnion(dataUnionAddress)
```
Comment (fonty1) [inline README.md:445]:
```suggestion
const dataUnion = new StreamrClient().getDataUnion(address)
```

File: README.md [hunk 7]
```diff
@@ -339,62 +341,87 @@ All the below functions return a Promise which gets resolved with the result.
...
-See `/docs` folder in this repo for Data Union `options` parameters. These docs can be rebuilt locally via:
+To deploy a new DataUnion:
+```js
+const dataUnion = await client.deployDataUnion()
 ```
-npm run docs
+
+To get an existing (previously deployed) `DataUnion` instance:
+```js
+const dataUnion = client.getDataUnion(address)
```
Comment (fonty1) [inline README.md:353]:
```suggestion
const dataUnion = client.getDataUnion(dataUnionAddress)
```

File: README.md [hunk 8]
```diff
@@ -339,62 +341,87 @@ All the below functions return a Promise which gets resolved with the result.
...
+| isMember(memberAddress)                                           | boolean             |                                                                             |
+| withdrawAll(\[[options](#withdraw-options)\])                     | Transaction receipt | Withdraw funds from Data Union                                              |
+| withdrawAllTo(recipientAddress\[, [options](#withdraw-options)\]) | Transaction receipt | Donate/move your earnings to recipientAddress instead of your memberAddress |
+| signWithdrawAllTo(recipientAddress)                               | Signature (string)  | Signature that can be used to withdraw all available tokens to given recipientAddress        |
+| signWithdrawAmountTo(recipientAddress, amountTokenWei)            | Signature (string)  | Signature that can be used to withdraw a specific amount of tokens to given recipientAddress |
 
-Here's an example how to sign off on a withdraw to (any) recipientAddress:
+Here's an example how to sign off on a withdraw to (any) recipientAddress (NOTE: this requires no gas!)
```
Comment (fonty1) [inline README.md:401]:
```suggestion
Here's an example on how to sign off on a withdraw to (any) recipientAddress (NOTE: this requires no gas!)
```

File: README.md [hunk 9]
```diff
@@ -413,19 +440,33 @@ These are available for everyone and anyone, to query publicly available info fr
...
+import { StreamrClient } from 'streamr-client'
 
-const withdrawableWei = await client.getMemberBalance(memberAddress)
+const dataUnion = new StreamrClient().getDataUnion(dataUnionAddress)
+const withdrawableWei = await dataUnion.getWithdrawableEarnings(memberAddress)
 ```
 
+### Withdraw options
+
+The functions `withdrawAll`, `withdrawAllTo`, `withdrawAllToMember`, `withdrawAllToSigned` all take an extra "options" argument. It's an object can contain following members:
```
Comment (fonty1) [inline README.md:451]:
```suggestion
The functions `withdrawAll`, `withdrawAllTo`, `withdrawAllToMember`, `withdrawAllToSigned` all take an extra "options" argument. It's an object that can contain the following parameters:
```

Comment (fonty1) [reply README.md:451]: Avoiding 'members' to avoid confusion with DU members

File: README.md [hunk 10]
```diff
@@ -534,6 +575,11 @@ In the browser, set `localStorage.debug = 'StreamrClient*'`
 
 Publishing to npm is automated via Github Actions. Follow the steps below to publish `latest` or `beta`.
 
+These docs can be rebuilt locally via:
```
Comment (fonty1) [inline README.md:578]:
```suggestion
For more technical documentation on the Data Unions API, see the [JS Client API Docs](https://streamr-dev.github.io/streamr-client-javascript/). These can also be rebuilt locally via:
```

File: README.md [hunk 11]
```diff
@@ -339,62 +341,87 @@ All the below functions return a Promise which gets resolved with the result.
...
 ```js
+import { StreamrClient } from 'streamr-client'
+
 const client = new StreamrClient({
     auth: { privateKey },
 })
 
 const dataUnion = await client.deployDataUnion()
-await client.setAdminFee(0.3, { dataUnion })
+const receipt = await dataUnion.setAdminFee(0.3, { dataUnion })
```
Comment (fonty1) [inline README.md:387]:
```suggestion
const receipt = await dataUnion.setAdminFee(0.3)
```

File: README.md (hunk 4 above)
Comment (fonty1) [reply README.md]: Fixed here

**Review by fonty1 — APPROVED** [review]

**Pull request merged**